docs/dev/topology-over-raft.md

```diff
@@ -104,19 +104,28 @@ that there are no tablet transitions in the system.
 Tablets are migrated in parallel and independently.
 
 There is a variant of tablet migration track called tablet draining track, which is invoked
-as a step of certain topology operations (e.g. decommission). Its goal is to readjust tablet replicas
+as a step of certain topology operations (e.g. decommission, removenode, replace). Its goal is to readjust tablet replicas
 so that a given topology change can proceed. For example, when decommissioning a node, we
 need to migrate tablet replicas away from the node being decommissioned.
 Tablet draining happens before making changes to vnode-based replication.
 
-# Tablet migration
+# Tablet transitions
 
-Each tablet has its own migration state machine stored in group0 which is part of the tablet state. It involves
+Tablets can undergo a process called "transition", which performs some maintenance action on the tablet which is
+globally driven by the topology change coordinator and serialized per-tablet. Transition can be one of:
+
+ * migration - tablet replica is moved from one shard to another (possibly on a different node)
+
+ * rebuild - new tablet replica is rebuilt from existing ones, possibly dropping old replica afterwards (on node removal or replace)
+
```

Comment:
In the future, we may want to make operations in different datacenters  (or even racks) independent. So we can migrate a tablet in dc1, and rebuild it in dc2. This won't work with this representation.

I don't propose fixing it now. I guess we could have had a separate transitions table to list individual transitions with better than tablet granularity. At least with raft changing schemas atomically is easier.